src/server/test/web/readingsLineGroupQuantity.js

```diff
@@ -547,6 +547,165 @@ mocha.describe('readings API', () => {
+                        },
+                        {
+                            // u12
+                            name: 'kg CO₂',
+                            identifier: '',
+                            unitRepresent: Unit.unitRepresentType.QUANTITY,
+                            secInRate: 3600,
+                            typeOfUnit: Unit.unitType.UNIT,
+                            suffix: 'CO₂',
+                            displayable: Unit.displayableType.ALL,
+                            preferredDisplay: false,
+                            note: 'special unit'
+                        },
+                        {
+                            // u13
+                            name: 'pound',
+                            identifier: 'lb',
+                            unitRepresent: Unit.unitRepresentType.QUANTITY,
+                            secInRate: 3600,
+                            typeOfUnit: Unit.unitType.UNIT,
+                            suffix: '',
+                            displayable: Unit.displayableType.ALL,
+                            preferredDisplay: false,
+                            note: 'special unit'
+                        }
+                    ];
+                    const conversionData = [
+                        //Add conversions c11, c12, c13, c14
+                        {
+                            // c11
+                            sourceName: 'Electric_Utility',
+                            destinationName: 'kg CO₂',
+                            bidirectional: false,
+                            slope: 0.709,
+                            intercept: 0,
+                            note: 'Electric_Utility → kg CO₂'
+                        },
+                        {
+                            // c12
+                            sourceName: 'kg CO₂',
+                            destinationName: 'kg',
+                            bidirectional: false,
+                            slope: 1,
+                            intercept: 0,
+                            note: 'CO₂ → kg'
+                        },
+                        {
+                            // c13
+                            sourceName: 'kg',
+                            destinationName: 'metric ton',
+                            bidirectional: true,
+                            slope: 1e-3,
+                            intercept: 0,
+                            note: 'kg → Metric ton'
+                        },
+                        { 
+                            // c14
+                            sourceName: 'pound', 
+                            destinationName: 'metric ton', 
+                            bidirectional: true, 
+                            slope: 454.545454, 
+                            intercept: 0, 
+                            note: 'lbs → metric tons' }
+                    ];
+
+                    // Redefine the meterData as the unit is different
```

Comment:
Even though the destination unit is different, the meter unit (Electric_Utility) is the same. Thus, you can use the standard meters in meterDatakWh & meterDataOther in readingsUtils.js along with groupDatakWh. It should be fine since the unitID below is for the correct graphing unit. Please let me know if this is not clear or you have issues.

Reply:
<img width="518" alt="image" src="https://github.com/OpenEnergyDashboard/OED/assets/117063757/217516d7-d9c0-404c-944f-146bb1dd0f0f">
Hi, I tried to use meterDatakWh & meterDataOther along with groupDatakWh in readingsUtils.js, but I will get the error in the image. I think it might be caused by the GraphicUnit defination in meterDatakWh & meterDataOther as "defaultGraphicUnit: 'kWh',".  In the redefined meterData & meterDataOther, we remove the "defaultGraphicUnit" so that the tests can pass.

Reply:
My bad. I overlooked that this test uses a direct conversion from Electric_Utility to kg CO2 and does not go via kWh. Thus, what you say is correct and the code was fine. I'm sorry for my oversight. The code is (and was) ready to merge.